src/gpg2john.c

```diff
@@ -820,10 +821,12 @@ string_to_key(void)
 		// hash_algs(Getc());
 		m_hashAlgorithm = Getc();
 		if (gpg_dbg) fprintf(stderr, "\t\tSalt - ");
-		if (gpg_dbg)
-			dump(8);
-		else
-			give(8, m_salt, sizeof(m_salt));
+		give(8, m_salt, sizeof(m_salt));
+		if (gpg_dbg) {
+			for (i = 0; i < 8; ++i) {
+				fprintf(stderr, "%02x", m_salt[i]);
+			}
+		}
 		if (gpg_dbg) fprintf(stderr, "\n");
```

Comment:
Oh, surely this `fprintf` should go into the `if` block above instead of checking the same condition again. I'm sorry I didn't notice this right away - it became more apparent after the previous cleanup.